be/src/storage/delta_writer.cpp

```diff
@@ -33,8 +33,8 @@
 
 namespace starrocks {
 
-OLAPStatus DeltaWriter::open(WriteRequest* req, MemTracker* mem_tracker, DeltaWriter** writer) {
-    *writer = new DeltaWriter(req, mem_tracker, StorageEngine::instance());
+OLAPStatus DeltaWriter::open(WriteRequest* req, MemTracker* mem_tracker, std::shared_ptr<DeltaWriter>& writer) {
```

Comment:
Coding specification: All parameters passed by reference must be added with const, otherwise pointers are used